config/initializers/secure_headers.rb

```diff
@@ -9,6 +9,10 @@
   connect_src = ["'self'", '*.newrelic.com', '*.nr-data.net', '*.google-analytics.com',
                  'services.assureid.net']
   connect_src << %w[ws://localhost:3035 http://localhost:3035] if Rails.env.development?
+  if LoginGov::Hostdata.in_datacenter? && Figaro.env.aws_region
+    image_upload_bucket_url = ImageUploadPresignedUrlGenerator.new.bucket_url
+    connect_src << "#{image_upload_bucket_url.gsub(/\/$/, '')}/*"
```

Comment:
A few ideas to clean this up, I don't feel strongly about either

1. `File.join` will add a slash between items if there isn't one, will leave it in if it's there. These aren't files per say....but the behavior is what we want I think

    ```suggestion
        connect_src << File.join(image_upload_bucket_url, '*')
    ```

2. `URI.join` is predictable if we *know* 1) the first argument has a protocol and 2) the second argument isn't an absolute path from root (_no_ leading slash on the second argument)

    ```suggestion
        connect_src << URI.join(image_upload_bucket_url, '*')
    ```

2. `String#chomp` can clean up the last character of a string if we're worried and want to do a simple concatenation

    ```suggestion
        connect_src << image_upload_bucket_url.chomp('/') + '/*'
    ```

Reply:
> 1. `File.join` will add a slash between items if there isn't one, will leave it in if it's there. These aren't files per say....but the behavior is what we want I think

I can't recall where exactly I'd read it, but I seem to remember reading something about how `File.join` can be a bit unreliable.

**Edit:** Ah, here it is: https://stackoverflow.com/a/34620369 . We're probably not overly concerned about Windows support, though it feels like asking for trouble.

> 2\. `URI.join` is predictable if we _know_ 1) the first argument has a protocol and 2) the second argument isn't an absolute path from root (_no_ leading slash on the second argument)

In my testing, this had the same result as in https://github.com/18F/identity-idp/pull/4409#discussion_r524529265:

```
$irb                                                                
irb(main):001:0> image_upload_bucket_url = 'https://s3.us-west-2.amazonaws.com/login-gov-idp-doc-capture-test.123456789-us-west-2/'
=> "https://s3.us-west-2.amazonaws.com/login-gov-idp-doc-capture-test.123456789-us-west-2/"
irb(main):002:0> URI.join(image_upload_bucket_url, '*').to_s
=> "https://s3.us-west-2.amazonaws.com/*"
```

> 3. `String#chomp` can clean up the last character of a string if we're worried and want to do a simple concatenation

Aha! I was looking for something like this, though I was looking for it as a variant of `rstrip` to customize the character removed. I'll update to use `chomp`.

Reply:
Chomping in 2be992a.